metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1932,6 +1939,83 @@ class MetalsLanguageServer(
           } yield ().asInstanceOf[Object]
         }
 
+      case ServerCommands.FormatThenRemoveBraces(textDocumentPositionParams) =>
+        CancelTokens.future { token =>
+          val uri = textDocumentPositionParams.getTextDocument().getUri()
+          val path = uri.toAbsolutePath
+          val initialFileCode = path.toInputFromBuffers(buffers).text
+
+          formattingProvider
+            .programmaticallyFormat(path, token)
+            .map(formattingEditsList =>
+              if (!formattingEditsList.isEmpty) {
```

Comment:
It's fine if it's empty.

Reply:
the method `programmaticallyFormat` returns empty if and only if scalafmt is not properly configured.

Reply:
Why just not check if the file exists

Reply:
I thought that checking whether scalafmt successfully finishes execution on the file, is a more certain way for making sure it is properly configured. And even if the file does not have any formatting problem, we still need the Scalafmt result for being able to conclude that..

Reply:
The logic works this way: we run scalafmt on the file, then continue the processing with that result regardless of whether it is different to the input or not.
If the scalafmt cannot produce a result, we do not continue the processing.